Test/Unit/Controller/Customer/DeleteCreditCardTest.php

```diff
@@ -0,0 +1,230 @@
+            ->disableOriginalConstructor()
+            ->setMethods(['create', 'load', 'delete','getId'])
+            ->getMock();
+
+        $this->requestMock = $this->getMockBuilder(Http::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getParam'])
+            ->getMock();
+
+        $this->messageMock = $this->getMockBuilder(Manager::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['addSuccessMessage','addExceptionMessage','addErrorMessage'])
+            ->getMock();
+
+        $this->validatorMock = $this->getMockBuilder(Validator::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['validate'])
+            ->getMock();
+
+        $this->redirectMock = $this->getMockBuilder(Redirect::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getRefererUrl'])
+            ->getMock();
+
+        $this->contextMock->method('getMessageManager')->willReturn($this->messageMock);
+        $this->contextMock->method('getRedirect')->willReturn($this->redirectMock);
+
+        $this->currentMock = $this->getMockBuilder(DeleteCreditCard::class)
+            ->setConstructorArgs([
+                $this->contextMock,
+                $this->bugsnagMock,
+                $this->customerCreditCardFactoryMock,
+                $this->validatorMock
+            ])->setMethods(['getRequest', '_redirect'])
+            ->getMock();
+    }
+
+
+    /**
+     * @test
+     */
+    public function execute_success()
+    {
+        $this->validatorMock->expects(self::once())->method('validate')->with($this->requestMock)->willReturn(true);
+
+        $this->requestMock->expects(self::once())->method('getParam')->willReturn(self::ID);
+        $this->messageMock->expects(self::once())
+            ->method('addSuccessMessage')
+            ->with(__('You deleted the Bolt credit card'))
+            ->willReturnSelf();
+        $this->redirectMock->expects(self::once())->method('getRefererUrl')->willReturnSelf();
+
+        $this->currentMock->expects(self::any())->method('getRequest')->willReturn($this->requestMock);
+        $this->currentMock->expects(self::once())->method('_redirect')->with($this->redirectMock)->willReturn($this->redirectMock);
+
+        $this->customerCreditCardFactoryMock->expects(self::once())->method('create')->willReturnSelf();
+        $this->customerCreditCardFactoryMock->expects(self::once())->method('load')->with(self::ID)->willReturnSelf();
+        $this->customerCreditCardFactoryMock->expects(self::once())->method('getId')->willReturn(self::ID);
+        $this->customerCreditCardFactoryMock->expects(self::once())->method('delete')->willReturnSelf();
+
+        $this->assertSame($this->redirectMock,$this->currentMock->execute());
+    }
+
+    /**
+     * @test
+     */
+    public function execute_withInvalidIdParameter(){
```

Comment:
(minor) please move opening curly brace to the next line to comply with PSR2 here and in methods below

Reply:
@oxigenka I updated the code. Can you re-check?